packages/flame/lib/src/sprite_batch.dart

```diff
@@ -171,29 +172,66 @@ class SpriteBatch {
     RSTransform? defaultTransform,
     Images? images,
     bool useAtlas = true,
+    bool needsFlips = true,
   }) async {
+    final _images = images ?? Flame.images;
     return SpriteBatch(
-      await _generateAtlas(images, path),
+      needsFlips
+          ? await _atlasFromCache(_images, path)
+          : await _images.load(path),
       defaultColor: defaultColor,
       defaultTransform: defaultTransform ?? RSTransform(1, 0, 0, 0),
       defaultBlendMode: defaultBlendMode,
       useAtlas: useAtlas,
     );
   }
 
-  static Future<Image> _generateAtlas(Images? images, String path) async {
-    final _images = images ?? Flame.images;
-    final image = await _images.load(path);
+  static const String _generatedAtlasKeySuffix = '#withFlipAttached';
+
+  static Future<Image> _atlasFromCache(Images images, String path) async {
+    if (!images.containsKey('$path$_generatedAtlasKeySuffix')) {
+      return images.addFuture(
+        '$path$_generatedAtlasKeySuffix',
+        _generateAtlas1(images, path),
+      );
+    }
+    return images.fromCacheFuture('$path$_generatedAtlasKeySuffix');
```

Comment:
Perhaps we should consider combining these into a single method, something like `Images.fetchOrGenerate(key, builder)`

Reply:
That is nice! - I really wanted something similar to how `.load` works but for images and this sounds like it could be that.

Reply:
That sounds good